inc/Engine/Optimization/LazyRenderContent/Frontend/Processor/Dom.php

```diff
@@ -28,6 +27,46 @@ class Dom implements ProcessorInterface {
 	 */
 	private $max_hashes;
 
+	/**
+	 * Array of patterns to exclude from hash injection.
+	 *
+	 * @since 3.17
+	 *
+	 * @var array
+	 */
+	private $exclusions = [];
+
+	/**
+	 * Sets the exclusions list
+	 *
+	 * @param string[] $exclusions The list of patterns to exclude from hash injection.
+	 *
+	 * @return void
+	 */
+	public function set_exclusions( $exclusions ): void {
+		$this->exclusions = $exclusions;
+	}
+
+	/**
+	 * Gets the exclusions pattern
+	 *
+	 * @return string
+	 */
+	private function get_exclusions_pattern(): string {
+		if ( empty( $this->exclusions ) ) {
+			return '';
+		}
+
+		$exclusions = array_map(
+			function ( $exclusion ) {
+				return preg_quote( $exclusion, '#' );
```

Comment:
I think we don't need to pass this `#` character, because we already use the `/` delimiter below in the `preg_match` call.